controllers/ansibleee_controller.go

```diff
@@ -72,6 +72,47 @@ func (r *AnsibleEEReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	//configMapVars := make(map[string]env.Setter)
 	//r.configMapForAnsibleEE(ctx, instance, helper, &configMapVars)
 
+	if instance.Spec.EnablePlugin == true {
+		if len(instance.Spec.Inventory > 0) {
+			fmt.Println("Using dynamic inventory, setting existing inventory to nil")
+			// do something with pre-existing inventory state?
+		}
+
+		foundSource := lookupInventorySource("./tmp/", "inventory_source.yml")
+
+		if foundSource == false {
+			fmt.Println("Source file is empty, cancelling job.")
+			// if empty need to specify inventory_plugin:
+			// ex) plugin: plugin_name
+		}
+
+		foundPlugin := &corev1.ConfigMap{}
+		err = r.Get(ctx, types.NamespacedName{Name: "plugin-configmap", Namespace: instance.Namespace, foundPlugin)
+
+		// store plugin source in config map variable
+		if != nil && errors.IsNotFound(err) {
+			//define new configmap to store plugin information
+			cm := r.createConfigMapPlugin(instance)
+			fmt.Printf("Creating a new ConfigMap: ConfigMap.Namespace %s ConfigMap.Name %s\n", cm.Namespace, "plugin-configmap")
+
+			err = r.Create(ctx, cm)
+			if err != nil {
+				fmt.Println(err.Error())
+				return ctrl.Result{}, err
+			}
+
+			fmt.Println("configmap created successfully - return and requeue")
+			return ctrl.Result{Requeue: true}, nil
+		} else if err != nil {
+			fmt.Println(err.Error())
+			return ctrl.Result{}, err
+		}
+
+		//execute job that loads plugin configmap into ./tmp/ dir
+		//run check_path method to verify path is not empty
+		//update instance.Spec.Inventory after plugin is ran?
```

Comment:
This will not be necessary, as the pod will not be reused afterwards: a new one will be created